content/chainguard/chainguard-images/videos/node-images.md

```diff
@@ -0,0 +1,291 @@
+in order to improve security and reduce image size."
+type: "article"
+date: 2024-04-25T15:21:01+00:00
+lastmod: 2024-04-25T15:21:01+00:00
+draft: false
+images: []
+menu:
+  docs:
+    parent: "chainguard-images"
+weight: 60
+toc: true
+---
+
+{{< youtube hfpVS-UP4Yw >}}
+
+## Tools used in this video
+
+* [Docker](https://docker.com)
+
+## Resources
+
+* Tutorial on [Porting a Sample Application](/chainguard/migration/porting-apps-to-chainguard/)
+* [Example Application Git repository](https://github.com/chainguard-dev/identidock-cg/) with code used in demo
+
+## Transcript
+
+Today, we're going to look at how easy it is to port a Dockerfile for a Node.js application to use
+a Chainguard image base and how it can help improve the image, especially in terms of security.
+
+I'll be using the free tier of Chainguard images here, so you can do everything in this video
+yourself today.
+
+OK, over to the terminal.
+
+So I have a directory here with a simple Node.js application.
+
+I think the first thing we should do is just see it running so you understand what it does.
+
+OK, so it's running on port 8080.
+
+If I do a curl, what we're going to do is curl port 8080.
+
+We're going to hit the monster endpoint and give it some input and a size.
+
+And what you'll see is we get an image back.
+
+So basically, it produces like an identicon, if you know what an identicon is, or a unique avatar
+for its input.
+
+So if I give it an input, so I'll give my manager Lisa's name here, and you get the appropriate
+avatar out.
+
+So we can put in anything we like here.
+
+And if you put the same thing in again, you should get the same thing back out.
+
+And it's sometimes used as a default image for users on a website, such as GitHub.
+
+OK, so let's take a look at the Dockerfile for this application.
+
+You can see that it uses the Node official image from a Docker Hub.
+
+It installs quite a few dependencies that are needed for the graphics capability, creates a non-root
+user, copies over the package JSON, and installs some Node modules.
+
+Then it copies over the code, changes to the dnmonster user, and calls `npm start`.
+
+So that all makes sense.
+
+Let's have a go at building it.
+
+And that was all cached, so that was really fast.
+
+But let's have a look at what it's built.
+
+So that image is 1.22 gigabytes in size.
+
+So it's a pretty large image.
+
+If we run it through Docker Scout, to look at the CVEs.
+
+I tend to use Docker Scout.
+
+It has quite nice summary output.
+
+But you can, of course, use Grype or whatever, et cetera.
+
+And what it's telling me is this image has two high, seven medium, and 109 low CVEs.
+
+It is saying that I can change to the `slim` image, and I will reduce that to three medium and 24
+low vulnerabilities.
+
+But we're not going to do either of that.
+
+We're going to change to use a Chainguard image and see how that affects things.
+
+So I'm going to use a Chainguard registry here at cgr.dev.
```

Comment:
I think it would be nice to link to the doc on the Registry

```suggestion
So I'm going to use the [Chainguard Registry](/chainguard/chainguard-registry/overview/) here at cgr.dev.
```